cg/apps/sequencing_metrics_parser/parsers/bcl_convert.py

```diff
@@ -0,0 +1,93 @@
+        self.sample_sheet_path: Path = bcl_convert_sample_sheet_file_path
+        self.run_info_path: Path = bcl_convert_run_info_file_path
+        self.quality_metrics: List[BclConvertQualityMetrics] = self.parse_metrics_file(
+            metrics_file_path=self.quality_metrics_path, metrics_model=BclConvertQualityMetrics
+        )
+        self.demux_metrics: List[BclConvertDemuxMetrics] = self.parse_metrics_file(
+            metrics_file_path=self.demux_metrics_path, metrics_model=BclConvertDemuxMetrics
+        )
+        self.adapter_metrics: List[BclConvertAdapterMetrics] = self.parse_metrics_file(
+            metrics_file_path=self.adapter_metrics_path, metrics_model=BclConvertAdapterMetrics
+        )
+        self.sample_sheet: List[BclConvertSampleSheetData] = self.parse_sample_sheet_file()
+        self.run_info: BclConvertRunInfo = self.parse_run_info_file()
+
+    def parse_metrics_file(
+        self, metrics_file_path, metrics_model: Callable
+    ) -> List[Union[BclConvertQualityMetrics, BclConvertDemuxMetrics, BclConvertAdapterMetrics]]:
+        """Parse specified BCL convert metrics file."""
+        LOG.info(f"Parsing BCLConvert metrics file: {metrics_file_path}")
+        parsed_metrics: List[
+            Union[BclConvertQualityMetrics, BclConvertDemuxMetrics, BclConvertAdapterMetrics]
+        ] = []
+        metrics_reader: List[Dict] = ReadFile.get_content_from_file(
+            file_format=FileFormat.CSV, file_path=metrics_file_path, read_to_dict=True
+        )
+        for sample_metrics_dict in metrics_reader:
+            parsed_metrics.append(metrics_model(**sample_metrics_dict))
+        return parsed_metrics
+
+    def get_nr_of_header_lines_in_sample_sheet(
+        self,
+    ) -> int:
+        """Return the number of header lines in a sample sheet.
+        Any lines before and including the line starting with [Data] is considered the header."""
+        csv_reader = ReadFile.get_content_from_file(FileFormat.CSV, self.sample_sheet_path)
+        header_line_count: int = 1
+        for line in csv_reader:
+            if SampleSheetHeaderColumnNames.DATA.value in line:
+                break
+            header_line_count += 1
+        return header_line_count
+
+    def parse_sample_sheet_file(self):
+        """Read in a sample sheet starting from the SAMPLE_SHEET_DATA_HEADER."""
+        LOG.info(f"Parsing BCLConvert sample sheet file: {self.sample_sheet_path}")
+        header_line_count: int = self.get_nr_of_header_lines_in_sample_sheet()
+        sample_sheet_list: List[BclConvertSampleSheetData] = []
+        with open(self.sample_sheet_path, "r") as sample_sheet_file:
+            for _ in range(header_line_count):
+                next(sample_sheet_file)
+            reader = csv.DictReader(sample_sheet_file)
+            for sample_sheet in reader:
+                sample_sheet_list.append(BclConvertSampleSheetData(**sample_sheet))
+        return sample_sheet_list
```

Comment:
These two functions will be deprecated with a new sample sheet reading module for bcl convert sample sheet. I will make this in a separate PR and then another one to update this class.